src/gui/base/WizardDialog.ts

```diff
@@ -206,14 +227,32 @@ class WizardDialogAttrs<T> {
 		if (finalAction) {
 			this.closeAction()
 		} else {
-			this.currentPage = currentIndex < lastIndex ? pages[currentIndex + 1] : pages[lastIndex]
+			this._goToPageAction(currentIndex < lastIndex ? currentIndex + 1 : lastIndex)
 		}
 	}
+
+	/** returns whether it is allowed to visit the page specified by pageIndex depending on selectedPageIndex */
+	allowedToVisitPage(pageIndex: number, selectedPageIndex: number): boolean {
+		if (pageIndex < 0 || selectedPageIndex < 0) {
+			return true // invalid values -> should not restrict here
+		}
+		const enabledPages = this._getEnabledPages()
+		// page is only allowed to be visited if it was already visited and there is no later page that was already visited and does not allow to go back
+		return (
+			pageIndex < selectedPageIndex &&
+			!enabledPages
+				.filter((page) => {
+					return enabledPages.indexOf(page) > pageIndex && enabledPages.indexOf(page) <= selectedPageIndex
```

Comment:
oooh, this is about ~O(3n²)

Reply:
I'm using the index from the filtering function now, which makes it linear I guess.